### PR TITLE
[Manual Backport 2.x][Feature] Flint query scheduler part1 - integrate job scheduler plugi…

### PR DESCRIPTION
**Pull request by noCharger**
### Description
Backport https://github.com/opensearch-project/sql/commit/3daf64fbce5a8d29e846669689b3a7b12c5c7f07 from https://github.com/opensearch-project/sql/pull/2834.

### CODE REVIEW

**Review by vamsi-amazon — APPROVED** [review]

File: async-query/src/main/java/org/opensearch/sql/spark/scheduler/OpenSearchAsyncQueryScheduler.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.index.engine.DocumentMissingException;
+import org.opensearch.index.engine.VersionConflictEngineException;
+import org.opensearch.jobscheduler.spi.ScheduledJobRunner;
+import org.opensearch.sql.spark.scheduler.job.OpenSearchRefreshIndexJob;
+import org.opensearch.sql.spark.scheduler.model.OpenSearchRefreshIndexJobRequest;
+import org.opensearch.threadpool.ThreadPool;
+
+/** Scheduler class for managing asynchronous query jobs. */
+public class OpenSearchAsyncQueryScheduler {
```
Comment (vamsi-amazon) [inline async-query/src/main/java/org/opensearch/sql/spark/scheduler/OpenSearchAsyncQueryScheduler.java:42]: I know I am late to the review. I should have asked these questions in main branch PR.

Can we have interfaces for Scheduler and other Parser class?

Comment (noCharger) [reply async-query/src/main/java/org/opensearch/sql/spark/scheduler/OpenSearchAsyncQueryScheduler.java:42]: Yeah part2 is gonna be there shortly which address this issue https://github.com/opensearch-project/sql/issues/2833

File: async-query/src/main/java/org/opensearch/sql/spark/scheduler/model/OpenSearchRefreshIndexJobRequest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import java.time.Instant;
+import lombok.Builder;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.jobscheduler.spi.schedule.Schedule;
+
+/** Represents a job request to refresh index. */
+@Builder
+public class OpenSearchRefreshIndexJobRequest implements ScheduledJobParameter {
```
Comment (vamsi-amazon) [inline async-query/src/main/java/org/opensearch/sql/spark/scheduler/model/OpenSearchRefreshIndexJobRequest.java:18]: can we extract generic things out of this into a async-query-core library?

File: async-query/src/main/java/org/opensearch/sql/spark/scheduler/OpenSearchAsyncQueryScheduler.java (hunk 1 above)
Comment (vamsi-amazon) [reply async-query/src/main/java/org/opensearch/sql/spark/scheduler/OpenSearchAsyncQueryScheduler.java:42]: I read comments in original review. It seems we are covered here.

File: async-query/src/main/java/org/opensearch/sql/spark/scheduler/model/OpenSearchRefreshIndexJobRequest.java (hunk 2 above)
Comment (vamsi-amazon) [reply async-query/src/main/java/org/opensearch/sql/spark/scheduler/model/OpenSearchRefreshIndexJobRequest.java:18]: Some part of this go to core library and it is covered.

**Review by vamsi-amazon — APPROVED** [review]

**Review by ykmr1224 — APPROVED** [review]

**Pull request merged**